doc/source/migration.rst

```diff
@@ -0,0 +1,105 @@
+.. _migration-guide:
+
+Migrating off of Eventlet
+=========================
+
+There are two main use cases for Eventlet:
+
+1. As a required networking framework, much like one would use ``asyncio``, ``trio``, or older frameworks like ``Twisted`` and ``tornado``.
+
+2. As an optional, pluggable backend that allows swapping out blocking APIs for an event loop, transparently, without changing any code.
+
+The second use case requires exact emulation of an ever-changing and ever-increasing API footprint, which is fundamentally unsustainable for a volunteer-driven open source project.
+This is why Eventlet is discouraging new users.
+If you really want to continue with this approach, you can use `gevent <https://www.gevent.org/>`_.
+But keep in mind that the same technical issues that make Eventlet maintenance unsustainable over the long term also apply to Gevent.
+
+**The rest of this document will focus on the first use case: Eventlet as the sole networking framework.**
+For this use case, we recommend migrating to Python's ``asyncio``, and we are providing infrastructure that will make this much easier, and allow for *gradual* migration.
+
+Step 1. Switch to the ``asyncio`` Hub
+-------------------------------------
+
+Eventlet has different pluggable networking event loops.
+By switching the event loop to use ``asyncio``, you enable running ``asyncio`` and Eventlet code in the same thread in the same process.
+
+To do so, set the ``EVENTLET_HUB`` environment variable to ``asyncio`` before starting your Eventlet program.
```

Comment:
I think we also need to speak about `eventlet.hubs.use_hub`.
This function allow to set the hub to use at runtime.

http://eventlet.net/doc/hubs.html#eventlet.hubs.use_hub

It would be an useful alternative for many of us.